### PR TITLE
Debugger - Adding functionality to variable explorer.

### PR DESCRIPTION
**Pull request by gushuro**
Adding functionality to the variable explorer in debug mode.

- Description and coloring of objects (such as Sprites).
- Expand/contract objects and arrays.
- Preview of arrays.

The coloring and design is subject to change.

### CODE REVIEW

**Comment by pelikhan** [comment]
Add screenshots or gif screen recordings.

File: pxtsim/langsupport.ts [hunk 1]
```diff
@@ -91,6 +91,23 @@ namespace pxsim {
...
             if (o.vtable && o.vtable.name) return o.vtable.name;
+            if (o instanceof RefCollection && (o as any).data) {
+                if ((o as any).data.length > 0) {
+                    let arrayType = typeof (o as any).data[0]
+                    switch (arrayType) {
+                        case "string":
+                        case "number":
+                        case "boolean":
+                            let contents = (o as any).data.slice(0,10).join(", ")
+                            return "[" + contents + "]";
```
Comment (pelikhan) [inline pxtsim/langsupport.ts]: if you drop items, you should show "...".

File: theme/debugger.less [hunk 2]
```diff
@@ -63,18 +63,34 @@
...
         font-weight: bold;
     }
     .variable .detail {
-        background:white;
+        // background:white;
+        // color: #87D282;
         color:black;
     }
+    .varval.number {
+        color: darkorchid; // TODO: Change this for the color we want for numbers
```
Comment (pelikhan) [inline theme/debugger.less]: define LESS variables in "site.variables" and use those

File: pxtsim/langsupport.ts (hunk 1 above)
Comment (gushuro) [reply pxtsim/langsupport.ts]: Yeah, I'm working on a clean way to do this.

**Comment by gushuro** [comment]
variable explorer v1 0

We have:
- Icons for expandable (and expanded) objects/arrays.
- Expansion of objects and arrays.
- Truncating long Strings and array's contents (and showing `...`)
- [To be changed] Colors for variables.

File: theme/debugger.less (hunk 2 above)
Comment (samelhusseini) [reply theme/debugger.less]: or define them at the top of the file, and have them reference standard lib colors in site.variables

Comment (samelhusseini) [reply theme/debugger.less]: but always use less variables.

**Comment by gushuro** [comment]
Closing as we'll be merging this changes in https://github.com/Microsoft/pxt/pull/5302.

**Pull request closed without merging**